R/DataWrangling.R

```diff
@@ -12,21 +12,26 @@ CleanData <- function(data) {
 #' @param data Data frame in which to search for treatment names
 #' @return Vector of all treatment names
 FindAllTreatments <- function(data) {
-  if ('T' %in% colnames(data)) {
-    # Long format
-    return(unique(data$T))
-  } else {
-    # Wide format
-    all_treatments <- c()
-    for (col in paste0('T.', seq(6))) {
-      if (col %in% colnames(data)) {
-        all_treatments <- c(all_treatments, data[[col]])
-      } else {
-        break
+  # Regular expression explanation:
+  # ^ = Start of string
+  # (?i) = Ignore case for matching
+  # (\\.[0-9]+)? = Optional group of full stop, followed by at least one digit
+  # $ = End of string
+  treatment_column_matches <- stringr::str_match(names(data), "^(?i)T(\\.[0-9]+)?$")
+  treatment_column_names <- treatment_column_matches[!is.na(treatment_column_matches)]
+  
+  treatment_names = unlist(
+    sapply(
+      treatment_column_names,
```

Comment:
I don't think you need unlist here, because sapply returns a vector. There are two other instances of this.

Reply:
Unfortunately removing the `unlist()` call causes errors. This seems to be because it's not guaranteed to be a vector. The return can be a matrix or list, depending on the output from the applied function.